src/cmd-diff

```diff
@@ -76,9 +76,19 @@ def main():
             active_differs += [differ]
 
     # ensure commits are imported if we know we'll need them
-    if any(differ.needs_ostree for differ in active_differs):
+    ostree_import = ''
+    for differ in active_differs:
+        if differ.needs_ostree == 'yes':
+            ostree_import = 'full'
+            # If we need a full import for one then we don't
+            # need to check the others.
+            break
+        elif differ.needs_ostree == 'partial':
+            ostree_import = 'partial'
+    if ostree_import != '':
```

Comment:
If we change it to an enum as suggested in the previous comment, then this could be e.g.

```
ostree_import = max([d.needs_ostree for d in active_differs])
if ostree_import > OstreeImport.NONE:
    ...
        ..., partial_import=(ostree_import == OstreeImport.PARTIAL))
```

Reply:
implemented an enum!